crates/resolver/src/name_server/name_server.rs

```diff
@@ -76,6 +76,16 @@ impl<P: ConnectionProvider> NameServer<P> {
         }
     }
 
+    // TODO: there needs to be some way of customizing the connection based on EDNS options from the server side...
+    pub(super) fn send(
+        &self,
+        request: DnsRequest,
+    ) -> Pin<Box<dyn Stream<Item = Result<DnsResponse, ProtoError>> + Send>> {
+        let this = self.clone();
+        // if state is failed, return future::err(), unless retry delay expired..
```

Comment:
This was a pre-existing comment but it seems a bit confusing to find here since the code below it doesn't do any specially handling of state or retry delays itself, it's handled by `inner.send()` - maybe it should be migrated to the headline of a non-rustdoc comment on that func?